temporalio/worker/_nexus.py

```diff
@@ -0,0 +1,467 @@
+        task completion.
+        """
+        # TODO(nexus-prerelease): headers
+        ctx = CancelOperationContext(
+            service=request.service,
+            operation=request.operation,
+            headers=headers,
+        )
+        _TemporalCancelOperationContext(
+            info=lambda: Info(task_queue=self._task_queue),
+            nexus_context=ctx,
+            client=self._client,
+        ).set()
+        try:
+            await self._handler.cancel_operation(ctx, request.operation_token)
+        except BaseException as err:
+            logger.warning("Failed to execute Nexus cancel operation method")
+            completion = temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                task_token=task_token,
+                error=await self._handler_error_to_proto(
+                    _exception_to_handler_error(err)
+                ),
+            )
+        else:
+            # TODO(nexus-preview): ack_cancel completions?
+            completion = temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                task_token=task_token,
+                completed=temporalio.api.nexus.v1.Response(
+                    cancel_operation=temporalio.api.nexus.v1.CancelOperationResponse()
+                ),
+            )
+        try:
+            await self._bridge_worker().complete_nexus_task(completion)
+        except Exception:
+            logger.exception("Failed to send Nexus task completion")
+
+    async def _handle_start_operation_task(
+        self,
+        task_token: bytes,
+        start_request: temporalio.api.nexus.v1.StartOperationRequest,
+        headers: Mapping[str, str],
+    ) -> None:
+        """
+        Handle a start operation task.
+
+        Attempt to execute the user start_operation method and invoke the data converter
+        on the result. Handle errors and send the task completion.
+        """
+
+        try:
+            start_response = await self._start_operation(start_request, headers)
+        except BaseException as err:
+            logger.warning("Failed to execute Nexus start operation method")
+            handler_err = _exception_to_handler_error(err)
+            completion = temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                task_token=task_token,
+                error=await self._handler_error_to_proto(handler_err),
+            )
+            if isinstance(err, concurrent.futures.BrokenExecutor):
+                self._fail_worker_exception_queue.put_nowait(err)
+        else:
+            completion = temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                task_token=task_token,
+                completed=temporalio.api.nexus.v1.Response(
+                    start_operation=start_response
+                ),
+            )
+
+        try:
+            await self._bridge_worker().complete_nexus_task(completion)
+        except Exception:
+            logger.exception("Failed to send Nexus task completion")
+        finally:
```

Comment:
Might be worth putting this try finally block around the entire function to ensure we can avoid leaks.

Reply:
Yes, I think that's a good call. I've done that. The activity worker can be improved in ths regard also (added to list).